docs/saved_objects/saved_object_access_control_design_proposal.md

```diff
@@ -0,0 +1,216 @@
+* Feature (API) access control is out of scope
+* Data (saved objects) access control is in scope
+
+## Design
+
+### Personas
+
+There are following personas in *Workspace* use cases:
+* **Super admin**: the administrator of the OpenSearch Dashboards, who has complete access to all functions and data of OpenSearch Dashboards.
+* **Workspace admin**: Workspace admin have complete access to workspace configurations and saved objects in the workspace. Workspace creator will become the workspace admin when a workspace is created. Workspace admin most time can be used exchangeably with and workspace owner. Workspace admin
+* **Workspace operator**: workspace operator has permission to view, create and update saved objects in the workspace.
+* **Workspace viewer**: workspace operator only has permission to view the saved objects in the workspace
+
+Please note that, the workspace admin, operator and viewer persona are specific to the workspace. e.g. one workspace operator in one workspace can be an admin in another workspace.
+
+super-admin is a stack level role, it can be configured in the `opensearch_dashboards.yml` config file.
+
+### Discretionary Access Control for Data Access Control
+
+In order to support allowing a resource owner to manage the access to the resource feature, we can employ discretionary access control model. The idea is that, for each OpenSearch Dashboards saved object, we can attach an ACL (access control list) to it to save the principals and permission to operate that saved object. When a user is attempting to perform any operation on that saved object, we will first of all evaluate the ACL and then allow or deny the operation accordingly.
+
+#### Why not using role based access control model
+
+In existing OpenSearch products, all access control follows role/attributes based access control model, where a user can have several roles, while each role has corresponding permissions. The issue with the role based access control model is that, it requires administrator to centrally manage all access policies, which defines what principals can perform what actions on what resources. With admin managed access policies, OpenSearch Dashboards users cannot grant other users permissions to access specific resources, nor can they add other users to new workspaces. Creating workspaces and managing user access to workspaces will become admin responsibility, which isn't a simple collaboration experience.
+
+### Workspace and Saved Objects
+
+Workspace is a newly introduced type of OpenSearch Dashboards data, it is a container which organize the functions and saved objects, so the saved object in a workspace inherit the permissions settings from the workspace. One saved object can potentially in multiple workspaces, thus it inherits the permission settings from all the workspaces it is assigned to.
```

Comment:
When you say inside, that seems more like a logical reference to 'associated with'.  I'm concerned about inheritance when there could be multiple workspace with conflicting rules.  As there are difference kinds of access read vs readwrite, if so this could make associating a saved object with multiple works spaces cause trouble - which permission do you get the lowest vs highest?

This inheritance creates side effects where modifying a workspace permission can change the permissions on objects inside other workspaces

Reply:
This phrase is missing a key verb:
>One saved object can potentially in multiple workspaces

can be in? can be referenced in?

I echo @peternied's sentiments about the undesired side effects. Will saved objects be addressable outside of a workspace, or will they always be nested inside of a workspace? i.e. if `object1` is associated with `workspace1` and `workspace2`, can I link to `object1` directly, or do I need to qualify with `workspace1/object1`? If so, could we skip the "it inherits all permissions across workspaces" so that it only gets those in the context it is being referenced?